index.html

```diff
@@ -233,42 +299,30 @@ <h1>🗺 Not on the map yet?</h1>
           })
           .then(
             function (response) {
-              const range = response.result;
+              const people = response.result.values;
 
-              var pinIcon = new google.maps.MarkerImage(
-                "./badger-pin.png",
-                null /* size is determined at runtime */,
-                null /* origin is 0,0 */,
-                null /* anchor is bottom center of the scaled image */,
-                new google.maps.Size(30, 50)
-              );
-
-              let lastInfoWindow;
-
-              range.values.forEach(function (row) {
+              // Initialise the markers
+              state.people = people.map(function (row, idx) {
                 const [name, lat, lng, slack] = row;
+                const position = { lat: +lat, lng: +lng };
 
                 const marker = new google.maps.Marker({
-                  position: { lat: +lat, lng: +lng },
+                  position,
                   animation: google.maps.Animation.DROP,
                   icon: pinIcon,
                   map,
                 });
-                const content = document.createElement("div");
-                content.className = "map-info-window";
-                content.innerHTML = "<h1></h1><p></p>";
-                content.querySelector("h1").textContent = name;
-                content.querySelector("p").textContent = `Slack: ${slack}`;
 
-                const infoWindow = new google.maps.InfoWindow({ content });
                 marker.addListener("click", function () {
-                  if (lastInfoWindow && lastInfoWindow.close) {
-                    lastInfoWindow.close();
+                  if (state.infoWindow && state.infoWindow.close) {
+                    state.infoWindow.close();
                   }
 
-                  lastInfoWindow = infoWindow;
-                  infoWindow.open(map, marker);
+                  state.infoWindow = infoWindow(idx);
+                  state.infoWindow.open(map, marker);
                 });
+
+                return { name, position, slack };
```

Comment:
I found `state.people = people.map(....)` slightly hard to read. I think the reason is that `infoWindow` is called visually  inside the map but is only passed in index. It might be easier to read if we get rid of `state.people` and pass `{ name, position, slack }` directly to `infoWindow` instead of the index.

Reply:
Yea, I can see how that would be confusing. Tried to clarify it with a bit of refactoring.

The issue is that we want to make the info window based on the *latest* state (incl. people data and location) at the the point the user clicks the marker, not at the point when the initial data fetch completed, which could be too early. The main issue this is solving right now is the race between fetching data and getting location from the browser. 

Later on, if we add any kind of live behaviour, it'll also keep the info windows up to date.